Sources/SwiftSyntax/SourceEdit.swift

```diff
@@ -13,24 +13,68 @@
 /// A textual edit to the original source represented by a range and a
 /// replacement.
 public struct SourceEdit: Equatable, Sendable {
-  /// The half-open range that this edit applies to.
-  public let range: Range<AbsolutePosition>
-  /// The text to replace the original range with. Empty for a deletion.
-  public let replacement: String
-
-  /// Length of the original source range that this edit applies to. Zero if
-  /// this is an addition.
-  public var length: SourceLength {
-    return SourceLength(utf8Length: range.upperBound.utf8Offset - range.lowerBound.utf8Offset)
+  /// The byte range of the original source buffer that the edit applies to.
+  public let range: ByteSourceRange
```

Comment:
I’d prefer to use `Range<AbsolutePosition>` here because I’m deprecating `ByteSourceRange` in https://github.com/apple/swift-syntax/pull/2588